### PR TITLE
Only care about istio components restart

### LINKED ISSUE #34345: Test RateLimiting failure
integ-telemetry-k8s-tests_istio  TestRateLimiting failed serveral times

https://gcsweb.istio.io/gcs/istio-prow/pr-logs/pull/istio_istio/34243/integ-telemetry-k8s-tests_istio/1419646370645544960/artifacts/telemetry-policy-63ff7a6781f34e/TestRateLimiting/_test_context/istio-ratelimit-2-88776-state838486265/cluster-0/

https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/34243/integ-assertion-k8s-tests_istio/1419646370691682304

The log is 

```
time="2021-07-26T13:18:23Z" level=debug msg="Implicit pipelining enabled: false"
panic: dial tcp 10.96.106.204:6379: connect: connection refused
```

From the redis log, redis started 3 seconds later than it

```
1:C 26 Jul 2021 13:18:26.408 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 26 Jul 2021 13:18:26.408 # Redis version=6.2.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 26 Jul 2021 13:18:26.408 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1:M 26 Jul 2021 13:18:26.409 * monotonic clock: POSIX clock_gettime
1:M 26 Jul 2021 13:18:26.410 * Running mode=standalone, port=6379.
1:M 26 Jul 2021 13:18:26.410 # Server initialized
1:M 26 Jul 2021 13:18:26.411 * Ready to accept connections
```

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Fix #34345

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/test/kube/dump.go [hunk 1]
```diff
@@ -244,10 +244,13 @@ func DumpPodLogs(_ resource.Context, c cluster.Cluster, workDir, namespace strin
 
 			// Get previous container logs, if applicable
 			if restarts := containerRestarts(pod, container.Name); restarts > 0 {
-				// This is only called if the test failed, so we cannot mark it as "failed" again. Instead, output
-				// a log which will get highlighted in the test logs
-				// TODO proper analysis of restarts to ensure we do not miss crashes when tests still pass.
-				scopes.Framework.Errorf("FAIL: pod %v/%v container %v restarted %d times", pod.Name, pod.Namespace, container.Name, restarts)
+				// only care about istio components restart
+				if container.Name == "istio-proxy" || container.Name == "discovery" {
```
Comment (howardjohn) [inline pkg/test/kube/dump.go:248]: I think we at least also care about istio-cni?

Comment (hzxuzhonghu) [reply pkg/test/kube/dump.go:248]: good, and also the init-container

File: pkg/test/kube/dump.go [hunk 2]
```diff
@@ -245,7 +245,8 @@ func DumpPodLogs(_ resource.Context, c cluster.Cluster, workDir, namespace strin
 			// Get previous container logs, if applicable
 			if restarts := containerRestarts(pod, container.Name); restarts > 0 {
 				// only care about istio components restart
-				if container.Name == "istio-proxy" || container.Name == "discovery" {
+				if container.Name == "istio-proxy" || container.Name == "discovery" || container.Name == "istio-init" ||
```
Comment (bianpengyuan) [inline pkg/test/kube/dump.go:248]: and `istio-validation` as well.

Comment (hzxuzhonghu) [reply pkg/test/kube/dump.go:248]: good

**Comment by hzxuzhonghu** [comment]
@howardjohn Is that ok to you

**Review by howardjohn — APPROVED** [review]

**Pull request merged**